### PR TITLE
chore(Python): added interpolation search file

### PR DESCRIPTION
**Pull request by RobHam99**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
Added interpolation search, in Python

**Other information:**

### CODE REVIEW

**Comment by cyberwizard1001** [comment]
Looks good to me. Follows python naming conventions and algorithm works for my testcases.

**Comment by cyberwizard1001** [comment]
@ming-tsai I am not a reviewer with write access.

**Comment by ming-tsai** [comment]
> @ming-tsai I am not a reviewer with write access. 

That doesn't affect you review the pull request

**Comment by RobHam99** [comment]
Do I need to do anything more?

**Review by cyberwizard1001 — APPROVED** [review]
Looks good

File: algorithms/Python/searching/interpolation_search.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    if isinstance(x, int) and isinstance(array, list): # check that x is integer and array is list
+        if len(array) < 1: # if array length is less than 1 print message
+            return False
+        elif len(array) == 1 and x in array: # if x is only element of array return index 0
+            return 0
+        elif len(array) > 1 and x in array: # search if array length > 1 and x is in the array
+            min = 0
+            max = len(array) - 1
+            while min <= max:
+                p = int(min + (x-array[min])*(max-min) / (array[max]-array[min]))
```
Comment (Arsenic-ATG) [inline algorithms/Python/searching/interpolation_search.py:14]: This formula will fail in the case when all the elements of the array are same ( i.e. `array[max]` == `array[min]` ) with  a `ZeroDivisionError`.

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
- [x] Handle the case where probe finding formula fails ( either by an additional conditional statement or by altering the formula itself ) 

- [x] Update the index readme file ( [DSA/algorithms/Python/README.md](https://github.com/MakeContributions/DSA/blob/main/algorithms/Python/README.md) ) of your fork with your creation.

File: algorithms/Python/searching/interpolation_search.py (hunk 1 above)
Comment (RobHam99) [reply algorithms/Python/searching/interpolation_search.py]: I have updated this

Comment (Arsenic-ATG) [reply algorithms/Python/searching/interpolation_search.py]: Shouldn't the output be consistant ?

the output of your new special case is (`min, max`) which is inconsistent compared to any other test case where it just returns only one value at where the search value is to be found.

Comment (RobHam99) [reply algorithms/Python/searching/interpolation_search.py]: but if array[max]==array[min] then there are two elements with the target value, so i thought i should return both indices. Is it better to only return one index?

Comment (Arsenic-ATG) [reply algorithms/Python/searching/interpolation_search.py]: consider the following cases 

---
**Case 1 :** 
```
arr = [1,1,1,1,1,1]
target_value = 1
```
your function returns `(0,5)`

---
**Case 2 :** 
```
arr = [0,0,0,1,1,1,1,1,1,2,3,4,5]
target_value = 1
```
your function returns `3`

---

You can see that although the answer is correct in both the cases, it shows inconsistency in output where for one exceptional case, the function is returning more than one values and for other it isn't.

either make the function return only one value, or handle the duplicates and return all the instances of the value found in the array ( both are fairly trivial tasks )

Comment (RobHam99) [reply algorithms/Python/searching/interpolation_search.py]: Ok I have made it only return the min index, is this done now?

**Review by Arsenic-ATG — APPROVED** [review]

**Pull request merged**